qiskit/pulse/schedule.py

```diff
@@ -337,14 +337,14 @@ def interval_filter(time_inst: Tuple[int, 'Instruction']) -> bool:
             return interval_filter
 
         filter_func_list = list(filter_funcs)
-        if channels:
+        if channels or channels == []:
```

Comment:
Ah! So that was the bug. Can we make these `if channels is not None`, etc? The issue is that `[]` evals to `False`. This mistake comes up a lot (I shouldve known better when I wrote this), for instance, when checking that an integer variable isn't `None`, but `0` also evals to `False`. So `not None` is the best comparison for all of these cases (rather than trying to enumerate the other values it could have which would eval to `False`).

Reply:
Ah cool, yeah `is not None` is way better :)